### PR TITLE
20240609 data parser

### PR DESCRIPTION
**Pull request by Jeffrey04**
In order to extract data out of the downloaded data (refer #1), run

```
$ legisdata extract 2020 2
```

This will extract the data with unstructured.io, and then pickled and saved. In order to parse the data into a structured format, 

```
$ legisdata parse 2020 2
```

The schema of the structured format can be found at `src/legisdata/schema.py`. The data in akomaNtoso format can then be found in each JSON file we generated above (`./data/2020/session-2/{hansard,inquiry}-parse/*.json`), by issuing

```
$ jq -r .akn $filename
```

if you have [batcat](https://github.com/sharkdp/bat) installed, you can pipe it over to have schema-highlighting

```
$ jq -r .akn $filename | batcat
```

As usual, data is posted to https://huggingface.co/datasets/sinarproject/legisdata/tree/main

### CODE REVIEW

**Comment by Jeffrey04** [comment]
if anyone is interested, you can go download a `.pickle` file from https://huggingface.co/datasets/sinarproject/legisdata and start experimenting with them. These are the notebooks that give you an idea on what the data look like

https://gist.github.com/Jeffrey04/b1737d5ee02a1ced0ac9c34b1a8fe827

(run in the same virtualenv as the project)

**Comment by Jeffrey04** [comment]
@sweemeng once mentioned unstructured.io requires a gpu to run, will try on another computer one day to test it

**Comment by Jeffrey04** [comment]
notes from @samqi 

On Github [sayit repo](https://github.com/mysociety/sayit/tree/master):  file [import_akomontoso.py ](https://github.com/mysociety/sayit/blob/master/speeches/importers/import_akomantoso.py#)uses debate:
debatesection records which [maps cleanly to 6.4 debate ](http://docs.oasis-open.org/legaldocml/akn-core/v1.0/os/part1-vocabulary/akn-core-v1.0-os-part1-vocabulary.html#_Toc523925090)structure
'debateSection', 'administrationOfOath', 'rollCall',
'prayers', 'oralStatements', 'writtenStatements',
'personalStatements', 'ministerialStatements',
'resolutions', 'nationalInterest', 'declarationOfVote',
'communication', 'petitions', 'papers', 'noticesOfMotion',
'questions', 'address', 'proceduralMotions',
'pointOfOrder', 'adjournment')
& containers ([speech, question, answer ](https://github.com/mysociety/sayit/blob/f60da55a85ba0870f920259b226fcc13aba7de37/speeches/importers/import_akomantoso.py#L225C31-L225C59)only)
full vocab has more but we[ ignore the full vocab as sayit only takes these 3](http://docs.oasis-open.org/legaldocml/akn-core/v1.0/os/part1-vocabulary/akn-core-v1.0-os-part1-vocabulary.html#_Toc523925090)

Thus contextually for
hansard 
would be  <debate name="hansard">  as per [open voices govhack-parliament.nz repo ](https://github.com/kayakr/parliament.nz/)which also made a [sayit compatible hansard xml for sayit](https://github.com/kayakr/parliament.nz/blob/7fd5642c87a6307c70e58b83a5b1cdcb5d5e6c50/data/akoma_ntoso_debate_nz_20170726_hand_crafted.xml#L3)
Example xml:
<debate name="hansard"><debatesection><pointOfOrder> <question> </question> ....<answer></answer>...
 inquiries,
 would have debatestructure soalan mulut & bertulis maps to oralStatements, writtenStatements accordingly
Example for soalan mulut/oral statements
 <debate><debatesection><oralStatements> <question> </question> ....<answer><answer>...

File: src/legisdata/main.py [hunk 1]
```diff
@@ -1,33 +1,93 @@
...
+
+class Meta(NamedTuple):
+    source: str
+    year: int
+    session: int
+    dun: str
+    parse_time: str = str(datetime.now())
+
+
+class Inquiry(NamedTuple):
```
Comment (Jeffrey04) [inline src/legisdata/main.py]: Just keeping as a note, this is the structure of the resulting JSON file for an inquiry

**Comment by samqi** [comment]
You can test GPU for free on Google Colab (. Use the T4 GPU  https://research.google.com/colaboratory/faq.html )

**Comment by Jeffrey04** [comment]
oh ya, forgot about it, but notebook is used mostly to test out ideas, and/or explore data
if you want to just read content in `.pickle` you don't need gpu

**Pull request merged**